lib/ical/recur_iterator.js

```diff
@@ -327,6 +327,16 @@ ICAL.RecurIterator = (function() {
         return null;
       }
 
+      // if there are no possible valid events return null.
+      try {
+        if (this.has_invalid_bysetpos()) {
+          return null;
+        }
+      } catch (err) {
+       // do nothing.
```

Comment:
does it make sense to continue if there was en exception thrown in `has_invalid_bysetpost` ?

Reply:
That code should never throw an error, this is just to make sure I don't accidentally break the parser if there's an edge case that isn't covered by the unit tests.